src/main/java/org/gwtbootstrap3/extras/toggleswitch/client/ui/base/ToggleSwitchBase.java

```diff
@@ -206,21 +209,59 @@ public void setValue(final Boolean value) {
         setValue(value, false);
     }
 
+    /**
+     * Called when this widget is attached to the DOM.
+     * 
+     * Here we want to make sure the state of the toggle switch hasn't changed before it was 
+     * attached or re-attached to the DOM.  If the value has changed update the toggle switch 
+     * via {@link #switchState(Element, boolean, boolean)}.
+     */
+    @Override
+    public void onAttach() {
+        super.onAttach();
+        boolean switchValue = switchState(getElement());
+        boolean value = getValue();
+        if (value != switchValue) {
+            switchState(getElement(), value, true);
+        }
+    }
+    
+    /**
+     * Sets the value of this widget.
+     * 
+     * If we are attached to the DOM, simply switch the state of to the new value and  
+     * if there is a switch in state then the onChange method will be called, if necessary, by 
+     * the 'switchChange.bootstrapSwitch' callback.
+     * 
+     * If we are not attached to the DOM, manually call the onChange method to update the value 
+     * of the underlying {@link CheckBox} and fire change events. 
+     */
     @Override
```

Comment:
I will revert all changes below. Your implementation is not correct here:
- first, when attaching to the DOM, the fireEvents is just never taken into account. The original implementation is much better and do exactly what you want
- second, the original ```onChange(boolean value)``` implementation is only called in:
```
private native void switchInit(Element e) /*-{
        $wnd.jQuery(e).bootstrapSwitch();
        var me = this;
        $wnd.jQuery(e).on('switchChange.bootstrapSwitch', function (em, state) {
            me.@org.gwtbootstrap3.extras.toggleswitch.client.ui.base.ToggleSwitchBase::onChange(Z)(state);
        });
    }-*/;
```
The change event is ALWAYS fired when user switch the value in the GUI. Thats why you cannot call ```ValueChangeEvent.fireIfNotEqual```, but ```ValueChangeEvent.fire```. Here, you don't need and is not supposed to compare the old value and the new one. When you want to set the toggle switch's value programmatically by calling ```switchState(Element e, boolean value, boolean skip)``` method, you can control whether or not the the change event should be fired with the 3rd argument ```skip```. So the original implementation (on the left side) was correct. You should just do nothing.